client/state/data-layer/wpcom-http/pipeline/retry-on-failure/index.jsx

```diff
@@ -0,0 +1,108 @@
+/**
+ * External dependencies
+ */
+import {
+	get,
+	head,
+	sortBy,
+	toPairs,
+} from 'lodash';
+
+const requestQueue = new Map();
```

Comment:
good points. because this is for flakey network issues I really wanted to encapsulate it in this middleware - make the Calypso code completely ignorant to it - meaning I didn't want to bleed into the actions themselves (actions shouldn't be coupled to the specifics of the particular implementation on the data provider, API, network, etc…)

when I started I though the different policies would need more than `retryCount` but in the end I think that's enough